main.py

```diff
@@ -301,59 +320,57 @@ def process_files(omr_files, template, args, out):
-        # literally)
-        finder = re.search(r'.*/(.*)/(.*)', filepath, re.IGNORECASE)
-        if(finder):
-            inputFolderName, filename = finder.groups()
-        else:
-            print("Error: Filepath not matching to Regex: " + filepath)
-            continue
-        # set global var for reading
 
-        inOMR = cv2.imread(filepath, cv2.IMREAD_GRAYSCALE)
-        print(
-            '\n[%d] Processing image: \t' %
-            (filesCounter),
-            filepath,
-            "\tResolution: ",
-            inOMR.shape)
+        filename = filepath.name
+        args['current_file'] = filepath
 
-        OMRCrop = utils.getROI(inOMR, filename, noCropping=args["noCropping"])
+        inOMR = cv2.imread(str(filepath), cv2.IMREAD_GRAYSCALE)
+        print('')
+        print(f'({filesCounter}) Opening image: \t{filepath}\tResolution: {inOMR.shape}')
+
+        # TODO: Get rid of saveImgList
+        for i in range(config.saveimglvl):
+            utils.resetSaveImg(i+1)
 
-        if(OMRCrop is None):
+        utils.appendSaveImg(1, inOMR)
+
+        # run preprocessors
+        for pp in template.preprocessors:
+            inOMR = pp.apply_filter(inOMR, args)           
+        
+        # Disable getROI for now
+        #OMRCrop = utils.getROI(inOMR, filename, noCropping=args["noCropping"])
+        
+        if(inOMR is None):
             # Error OMR - could not crop
             newfilepath = out.paths.errorsDir + filename
             out.OUTPUT_SET.append([filename] + out.emptyResp)
             if(checkAndMove(config.NO_MARKER_ERR, filepath, newfilepath)):
                 err_line = [filename, filepath,
                             newfilepath, "NA"] + out.emptyResp
-                pd.DataFrame(
-                    err_line,
-                    dtype=str).T.to_csv(
-                    out.filesObj["Errors"],
-                    quoting=QUOTE_NONNUMERIC,
-                    header=False,
-                    index=False)
+                pd.DataFrame(err_line, dtype=str) \
+                  .T                              \
+                  .to_csv(out.filesObj["Errors"],
+                          mode='a',
+                          quoting=QUOTE_NONNUMERIC,
+                          header=False,
+                          index=False)
             continue
 
-        if template.marker is not None:
-            OMRCrop = utils.handle_markers(OMRCrop, template.marker, filename)
+        # resize to conform to template
+        imOMR = utils.resize_util(inOMR, config.uniform_width, config.uniform_height)
 
         if(args["setLayout"]):
             templateLayout = utils.drawTemplateLayout(
-                OMRCrop, template, shifted=False, border=2)
+                inOMR, template, shifted=False, border=2)
             utils.show("Template Layout", templateLayout, 1, 1)
             continue
 
         # uniquify
-        file_id = inputFolderName + '_' + filename
+        file_id = str(filename)
```

Comment:
filename may not be unique across directories

Reply:
The output now follows the same directory structure as the input. So there will not be an issue with files of same names as they will be in their own directory